Widgets/StackedCodeWidget.cpp

```diff
@@ -0,0 +1,56 @@
+#include "StackedCodeWidget.h"
+#include "CodeWidget.h"
+
+#include <QFileDialog>
+#include <QInputDialog>
+#include <QMessageBox>
+#include <QTextStream>
+
+StackedCodeWidget::StackedCodeWidget(QWidget* parent) :  QStackedWidget(parent) {}
+
+void StackedCodeWidget::newSource() {
```

Comment:
Does this happen a lot? Consider extracting a template that delegates methods to its type.